lib/logstash/inputs/beats.rb

```diff
@@ -131,44 +131,40 @@ def register
       LogStash::Logger.setup_log4j(@logger)
     end
 
-    if !@ssl
-      @logger.warn("Beats input: SSL Certificate will not be used") unless @ssl_certificate.nil?
-      @logger.warn("Beats input: SSL Key will not be used") unless @ssl_key.nil?
-    elsif !ssl_configured?
-      raise LogStash::ConfigurationError, "Certificate or Certificate Key not configured"
-    end
+    if @ssl
+      if @ssl_key.nil? || @ssl_key.empty?
+        configuration_error "Missing required setting ssl_key => ... (due ssl => true)"
```

Comment:
Maybe could be reworded a little?

```suggestion
        configuration_error "ssl_key => is a required setting when ssl=>true is configured"
```

Reply:
thanks, fine by me - what I was trying is to be a bit similar to the generic (required) parameter validation:
```
Missing a required setting for the beats input plugin:

  input {
    beats {
      port => # SETTING MISSING
      ...
    }
  }
```